### ISSUE TITLE
[bug] missing schema definition from Federated _service { sdl } query

### ISSUE THREAD

**Issue by dariuszkuc** [hayes/pothos#663]
Hello 👋 
While updating [Apollo Federation subgraph compatibility tests](https://github.com/apollographql/apollo-federation-subgraph-compatibility/pull/262) I've noticed that Pothos implementation currently does not expose `@link` directive on the schema/schema extension definition. Our previous test was incorrectly reporting `@link` support based on the presence of `@link` string in the schema (SDL contains link directive definition which satisfied the test).

Expected behavior:
- sdl should include `schema/schema extension` definition with `@link` directive applied

```graphql
extend schema
  @link(
    url: "https://specs.apollo.dev/federation/v2.0",
    import: ["@extends", "@external", "@inaccessible", "@key", "@override", "@provides", "@requires", "@shareable", "@tag"]
  )
```

**Comment by hayes** [comment]
@dariuszkuc thanks for the report.  Looks like there is an issue in the `printSubgraphSchema` function that prevents this from being implemented properly.

There is a branch here for non-core schemas: https://github.com/apollographql/federation/blob/main/internals-js/src/federation.ts#L1093  which calls this function that only works with core schemas: https://github.com/apollographql/federation/blob/main/internals-js/src/federation.ts#L1167

Basically, if you have a @link directive in your AST, it assumes that the schema is a core schema, and I don't see an obvious way to generate the required schema.  I think this branch of the `completeSubgraphSchema` function needs to be updated
